added quickstart section to documentation (#117)
esm-tools/pymorize: doc/quickstart.rst

===========
Quick start
===========

Installation
------------

Installation from source repository::

  git clone https://esm-tools/pymorize.git
  cd pymorize
  pip install pymorize[<extras>]

> For more detials in installation options, please refer section installation_

Reviewer: ```suggestion
For more details in installation options, please refer section installation_
```

Setting up a task to pymorize
-----------------------------

> At the heart of pymorize is the yaml configuration file. It gathers all
> the information it needs to perform tasks from the configuration file.

Reviewer: ```suggestion
At the heart of ``pymorize`` is the yaml configuration file. ``pymorize`` gathers all
the information it needs to perform CMORization of your data from this file.
```

The yaml file has 4 sections:
- ``general`` global settings that are applicable to all the rules
- ``pymorize`` setting for the controlling the behavior of the tool
- ``rules`` each rule defines parameters per variable.
- ``pipelines`` processing steps to carry out cmorization procress.

For detailed description on this sections, please refer to pymorize_building_blocks_

> As an example task to cmorize ``CO2f`` variable, create a file called ``basic.yaml`` and populate with the following content

Reviewer: ```suggestion
As an example task to cmorize ``FESOM 1.4``'s ``CO2f`` variable, create a file called ``basic.yaml`` and populate with the following content
```

      CV_Dir: /Users/pasili001/repos/pymorize/cmip6-cmor-tables/CMIP6_CVs
    pymorize:
      warn_on_no_rule: False
      dask_cluster: "local"
      enable_output_subdirs: False
    rules:
      - name: process_CO2f
        inputs:
          - path: /Users/pasili001/sampledata
            pattern: CO2f_fesom_.*nc
        cmor_variable: fgco2
        model_variable: CO2f
        output_directory: .
        variant_label: r1i1p1f1
        experiment_id: piControl
        source_id: AWI-CM-1-1-HR
        model_component: seaIce
        grid_label: gn
        pipelines:
          - default
    pipelines:
      - name: default
        steps:
          - "pymorize.gather_inputs.load_mfdataset"
          - "pymorize.generic.get_variable"
          - "pymorize.timeaverage.compute_average"
          - "pymorize.units.handle_unit_conversion"
          - "pymorize.global_attributes.set_global_attributes"
          - "pymorize.generic.trigger_compute"
          - "pymorize.files.save_dataset"
      - name: partial
        steps:
          - "pymorize.gather_inputs.load_mfdataset"
          - "pymorize.generic.get_variable"
          - "pymorize.units.handle_unit_conversion"

> Here is a breif description of each field in each section.

Reviewer: ```suggestion
Here is a brief description of each field in each section.
```

        model_component: |
        grid_label:      |
        pipelines:
          - default <- which pipeline to use. (choose default or partial)
    pipelines:
      - name: default  <- any descriptive name
        steps:
          - "pymorize.gather_inputs.load_mfdataset"
          - "pymorize.generic.get_variable"
          - "pymorize.timeaverage.compute_average"
          - "pymorize.units.handle_unit_conversion"
          - "pymorize.global_attributes.set_global_attributes"ß
          - "pymorize.generic.trigger_compute"
          - "pymorize.files.save_dataset"
      - name: partial
        steps:
          - "pymorize.gather_inputs.load_mfdataset"
          - "pymorize.generic.get_variable"
          - "pymorize.units.handle_unit_conversion"

There is more that can be specified in the configuration file but for
now this is good enough to get started.

Before running the task, it should be possible to validate the config
for a sanity check as follows

.. code:: shell

  ➜ pymorize validate config basic.yaml

To run the task just run the following command

.. code:: shell

  ➜ pymorize process basic.yaml

As the tool is working on the task, a lot of logging information is

> spitted out to the terminal screen. The same information is also written

Reviewer: ```suggestion
printed out to the terminal screen. The same information is also written
```

- checking unit conversion:
  In this example, model variable ``CO2f`` has
  units ``mmolC/m2/d``. The cmor variable ``fgco2`` has units
  ``kg m-2 s-1``. This means there needs to be a conversion factor to
  express moles of Carbon in grams. Pymorize detects such units and
  applies the appropriate unit conversion factor. Search for ``molC``
  in the logs

  .. code:: shell

    ➜ grep -i "molC" $(ls -rthd logs/pymorize-process* | tail -n 1 )
    2025-03-13 09:06:37.158 | INFO     | pymorize.units:handle_unit_conversion:148 - Converting units: (CO2f -> fgco2) mmolC/m2/d -> kg m-2 s-1 (kg m-2 s-1)
    2025-03-13 09:06:37.158 | DEBUG    | pymorize.units:handle_chemicals:67 - Chemical element Carbon detected in units mmolC/m2/d.
    2025-03-13 09:06:37.158 | DEBUG    | pymorize.units:handle_chemicals:68 - Registering definition: molC = 12.0107 * g
    2025-03-13 09:06:37.470 | INFO     | pymorize.units:handle_unit_conversion:148 - Converting units: (CO2f -> fgco2) mmolC/m2/d -> kg m-2 s-1 (kg m-2 s-1)

Hopefully, this is good enough as a starting point for using this tool.

As next steps checkout ``examples`` directory for ``sample.yaml`` file which

> contains more configuarion options and also ``pymorize.slurm`` file which is

Reviewer: ```suggestion
contains more configuration options and also ``pymorize.slurm`` file which is
```
